enhancements/ingress/aws-load-balancer-operator.md

```diff
@@ -317,43 +318,48 @@ does not remove the existing addon attachment from the provisioned load
+the `cloud-credentials-operator` will be running in `manual mode`, the credentials
+for the lb-controller won't be provisioned automatically, and will require manual
+intervention to do so. The required manual steps will be documented within the
+operator docs. Additional documentation for STS mode/manual mode can be found [here](https://docs.openshift.com/container-platform/4.10/authentication/managing_cloud_provider_credentials/cco-mode-sts.html)
+
+The lb-controller requires a validating and mutating webhook for correct operation.
+The operator will have to create the webhooks along with the controller deployment. 
+The webhook can be registered with a CA bundle which is used to verify the identity 
+of webhook by the API server. The [service-ca controller](https://docs.openshift.com/container-platform/4.10/security/certificate_types_descriptions/service-ca-certificates.html)
+can be used to generate certificates and have them injected into the webhook
+configurations. The operator does not manage the lifecycle of the certificates
+in the associated secret. Additional details mentioned in [Drawbacks](#drawbacks).
 
 ### Risks and Mitigations
 
-#### Restricting features in the lb-controller
-
-Some changes will have to be made in the lb-controller to ensure that the users
-only use features which are supported on OCP and only those features which
-target Ingress resources. These changes may also be contributed back upstream
-since it may benefit other users.
-
-1. The annotation `alb.ingress.kubernetes.io/target-type` can be set to the
-value **_ip_** or **_instance_**. When the value is set to _instance_ the
-created load balancer will be configured to route traffic through a NodePort
-service. The _ip_ value only works when the cluster uses the Amazon VPC CNI and
-hence will not work on OCP/OKD clusters. The controller has to be modified so
-that this annotation is ignored and the controller always uses routing through
-_NodePort_ services.
-2. The _lb-controller_ has support for _Service_ type resources as
-well. When the annotation `service.beta.kubernetes.io/aws-load-balancer-type:
-"external"` is specified on a _Service_ resource of type _LoadBalancer_ the
-in-tree controller ignores this resource and the lb-controller instead
-provisions a Network Load Balancer with the correct configuration. Since we want
-to restrict the features supported by the controller to only Ingresses the
-lb-controller will have to be modified so that it does not reconcile any Service
-resources.
+The annotation `alb.ingress.kubernetes.io/target-type` can be set to the
+value **_ip_** or **_instance_**. When the value is set to _instance_ (default)
+the created load balancer will be configured to route traffic through a NodePort
+service. Setting the annotation to _ip_ is not huge risk, since it only works when
+the cluster uses the Amazon VPC CNI and hence will not work on OCP/OKD clusters.
```

Comment:
I'm confused.  Is the risk that the user might set this annotation to a problematic value?  If the user does set it to "ip", what happens?  (What does "will not work mean"?)  Is this what the `EnableIPTargetType` feature gate was added to prevent?  
```suggestion
The user can set the `alb.ingress.kubernetes.io/target-type` annotation to the
value **_ip_** or **_instance_**. When the value is set to _instance_,
the created load balancer will be configured to route traffic through a NodePort
service.  This is also the default behavior when the annotation is not set.
However, when the value is set to _ip_, the lb-controller requires that
the cluster use the Amazon VPC CNI, and hence [the lb-controller
 will not do anything|the lb-controller will behave the same as if
 "instance" were specified|something else?] on OCP/OKD clusters.
```

Reply:
If the user does set the value to `ip`, the controller errors since the `TargetGroupBinding` cannot resolve pod ENI since it is dependent on AWS VPC CNI. 
>`
{"level":"error","ts":1654582687.4895403,"logger":"controller-runtime.manager.controller.targetGroupBinding","msg":"Reconciler error","reconciler group":"elbv2.k8s.aws","reconciler kind":"TargetGroupBinding","name":"k8s-awsloadb-echoserv-362ca47f2d","namespace":"aws-load-balancer-test-custom-ing","error":"cannot resolve pod ENI for pods: [aws-load-balancer-test-custom-ing/echoserver]"}
`

Yes, `EnableIPTargetType` was introduced to prevent this. It is yet to be released upstream, hence the current risk.

Reply:
`EnableIPTargetType` is carried in the downstream fork though, right?  Anyway, can you update the text to clarify the issues raised in my earlier comment?